src/OrchardCore.Modules/OrchardCore.Contents/Indexing/ContentItemIndexCoordinator.cs

```diff
@@ -49,7 +51,9 @@ public async Task BuildIndexAsync(BuildIndexContext context)
                 var partActivator = _contentPartFactory.GetTypeActivator(partTypeName);
                 var part = (ContentPart)context.ContentItem.Get(partActivator.Type, partName);
 
-                var typePartIndexSettings = contentTypePartDefinition.GetSettings<ContentIndexSettings>();
+                MethodInfo ContentTypePartDefinitionMethod = typeof(ContentTypePartDefinition).GetMethod("GetSettings");
```

Comment:
No need of generics, just access the property using the settings type name sub-property.